test/sequential/test-fs-watch.js

```diff
@@ -124,41 +124,45 @@ function repeat(fn) {
 // Whitebox test to ensure that wrapped FSEvent is safe
 // https://github.com/joyent/node/issues/6690
 {
-  let oldhandle;
-  assert.throws(
-    () => {
-      const w = fs.watch(__filename, common.mustNotCall());
-      oldhandle = w._handle;
-      w._handle = { close: w._handle.close };
-      w.close();
-    },
-    {
-      name: 'Error',
-      code: 'ERR_INTERNAL_ASSERTION',
-      message: /^handle must be a FSEvent/,
-    }
-  );
-  oldhandle.close(); // clean up
+  if (common.isOSX || common.isWindows) {
+    let oldhandle;
+    assert.throws(
+      () => {
+        const w = fs.watch(__filename, common.mustNotCall());
+        oldhandle = w._handle;
+        w._handle = { close: w._handle.close };
+        w.close();
+      },
+      {
+        name: 'Error',
+        code: 'ERR_INTERNAL_ASSERTION',
+        message: /^handle must be a FSEvent/,
+      }
+    );
+    oldhandle.close(); // clean up
+  }
 }
 
 {
-  let oldhandle;
-  assert.throws(
-    () => {
-      const w = fs.watch(__filename, common.mustNotCall());
-      oldhandle = w._handle;
-      const protoSymbols =
-        Object.getOwnPropertySymbols(Object.getPrototypeOf(w));
-      const kFSWatchStart =
-        protoSymbols.find((val) => val.toString() === 'Symbol(kFSWatchStart)');
-      w._handle = {};
-      w[kFSWatchStart]();
-    },
-    {
-      name: 'Error',
-      code: 'ERR_INTERNAL_ASSERTION',
-      message: /^handle must be a FSEvent/,
-    }
-  );
-  oldhandle.close(); // clean up
+  if (common.isOSX || common.isWindows) {
+    let oldhandle;
+    assert.throws(
+      () => {
+        const w = fs.watch(__filename, common.mustNotCall());
+        oldhandle = w._handle;
+        const protoSymbols =
+          Object.getOwnPropertySymbols(Object.getPrototypeOf(w));
+        const kFSWatchStart =
+          protoSymbols.find((val) => val.toString() === 'Symbol(kFSWatchStart)');
+        w._handle = {};
+        w[kFSWatchStart]();
+      },
+      {
+        name: 'Error',
+        code: 'ERR_INTERNAL_ASSERTION',
+        message: /^handle must be a FSEvent/,
+      }
```

Comment:
Shouldn't we align with that? Alternatively, we could `[kFSWatchStart]` method in the JS implementation, because we don't really need it after all, it could all happen in the constructor, no?

Reply:
Since this is an undocumented variable `watcher._handle`, I think we are OK with omitting it instead of aligning with it. I think we should move away from `._handle` in the future when we have the luxury to introduce a breaking change.

Reply:
What this test is doing it trying to call `FSWatcher.prototype[kFSWatchStart]` on an object that is not a `FSWatcher` instance, it has little to do with `_handle`. But I think the question remains: why do we bother with a `[kFSWatchStart]` method? If it's to maximize compatibility with the native implementation, then I think it should also pass this test.

Reply:
This test assumes that `kFSWatchStart` calls `_handle` and throws an error if `_handle` is not an FSEvent. Since non-native implementation does not call `_handle` or validates `handle` being a FSEvent, I did not understand the reasoning behind this.

We are using `kFSWatchStart` because making it private also makes it unavailable to the `lib/fs.js`, only way to call a function outside the defined class, and not expose it to the userland (which is available through the usage symbols).

Reply:
FWIW the function is still exposed to userland (e.g. this test can access it), it makes it harder to access it and serves as a signal to say "this is internal, please don't use it".
Every time this function is called, it's right after calling the constructor. Why do we need a separate function? Why not copy the code inside of `[kFSWatchStart]` and put it in the constructor?

Reply:
I made the that choice to create the same interface as the native implementation. For an outside collaborator, it would be much easier to maintain & navigate through the code, and sync the implementations between native and non-native implementations.